### ISSUE TITLE
tag_deck.py not tagging decks

### ISSUE THREAD

**Issue by cmahlen** [zachalmers/Anki_Tagger#3]
Thanks for putting this together, it's a really great idea and I'm glad you're sharing it with everyone. 

I've been able to generate a .csv with all the cards from the learning guide, but I'm having trouble with the last step where `tag_deck.py` is supposed to take this .csv and tag an .apkg file. I've never manipulated .apkg files from a python program before, so I'm not really sure how this is supposed to work. When I run `tag_deck.py <cards.csv> <deck.apkg>`, there are no errors, but nothing happens.

Any help would be greatly appreciated.

**Comment by zachalmers** [comment]
It basically works as expected. It's manipulating the underlying database
file of the apkg to insert the tags. Did you have a specific question?

**Comment by cmahlen** [comment]
Yes, I should have also added that I can't see any new tags when I import the deck into Anki.

**Comment by zachalmers** [comment]
Hmmm. I'm not sure but if you made it this far you should be able to debug
going thru the code.

**Comment by jonathan-kuo11** [comment]
This seems to be the same error (or a similar error) faced in [this reddit thread](https://www.reddit.com/r/Anki/comments/ntrv93/marking_cards_for_sync_after_modifying_the/) and documented in [AnkiPandas](https://github.com/klieret/AnkiPandas/issues/137). That is, as described in [the documentation](https://addon-docs.ankiweb.net/the-anki-module.html), using `col.db.execute` will not force the database to sync (i.e., update existing cards, despite the presence of the new tags on the cards). I have verified that tags are being added to `anki_deck.apkg` by inspecting the tags field in SQLite for the addition of tags after running `tag_deck.py`.

To fix this, you can update the `mod` field to the current epoch time (at script run) using the `time` module. This will indicate to anki that the card has been edited since export and needs to be refreshed with new data. Will put in fix shortly. I'm not 100% sure this works, as I think there may have been a discrepancy between number of cards tagged vs. number of cards updated, but will figure it out.
